presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizerInformationCollector.java

```diff
@@ -15,11 +15,16 @@
 
 import com.facebook.presto.spi.eventlistener.PlanOptimizerInformation;
 
+import javax.annotation.concurrent.GuardedBy;
+import javax.annotation.concurrent.ThreadSafe;
+
 import java.util.LinkedList;
 import java.util.List;
 
+@ThreadSafe
 public class OptimizerInformationCollector
 {
+    @GuardedBy("this")
```

Comment:
Hmm is this better or do we use Atomic

Reply:
this doesn't actually make it synchronized, it's just an annotation, you would need to also make all the access methods synchronized.  Alternatively, you can use a ConcurrentLinkedQueue.  In either case, I would make an immutableCopy before returning in getOptimizationInfo().  otherwise the caller can modify the object.

Reply:
> Hmm is this better or do we use Atomic

Since the OptimizerInformationCollector is per session object, we should not hit concurrency issue. Use synchronized method to be consistent to warning collector here.